app/controllers/api/v1/items_controller.rb

```diff
@@ -33,6 +33,25 @@ def destroy
     item.destroy
   end
 
+  def find_all
+    if params[:min_price].present? || params[:max_price].present?
+      items = Item.find_items_by_price(params.slice(:min_price, :max_price))    
+    elsif params[:name].present?
+      items = Item.find_items_by_name(params[:name])
+    else
+      render json: { error: "No search parameters provided" }, status: :bad_request
+      return
+    end
+  
```

Comment:
I like the logic here. We should be able to handle errors with the errors serializer, but we can mess with that later.